src/components/Button/Button.tsx

```diff
@@ -1,49 +1,186 @@
+        ],
+      },
+      {
+        color: 'green',
+        variant: 'filled',
+        class: [
+          'bg-green-700',
+          'text-text-50',
+          'hover:border-green-800',
+          'hover:bg-green-600',
+          'hover:text-green-50',
+          'disabled:border-green-900/80',
+          'disabled:bg-green-900/60',
+          'disabled:text-green-900',
+          'disabled:hover:border-green-900/80',
+          'disabled:hover:bg-green-900/60',
+        ],
+      },
+    ],
+    defaultVariants: {
+      variant: 'outline',
+      size: 'md',
+      color: 'base',
+    },
+  },
+);
+
+export const iconSizes = {
+  sm: 'size-3',
+  md: 'size-5',
+  lg: 'size-6',
+  xl: 'size-7',
+};
+
+interface ButtonProps
+  extends Omit<React.ButtonHTMLAttributes<HTMLButtonElement>, 'color'>,
+    VariantProps<typeof buttonVariants> {
+  asChild?: boolean;
+  startIcon?: React.ReactNode;
+  endIcon?: React.ReactNode;
+  type?: 'button' | 'submit' | 'reset';
   round?: boolean;
+  icon?: boolean;
+  loading?: boolean;
 }
 
-const Button: React.FC<PropsWithChildren<ButtonProps>> = ({
-  onClick,
-  tooltip,
-  className,
-  filled,
-  children,
-  ghost,
-  red,
-  round,
-  ...props
-}) => {
-  return (
-    <button
-      onClick={() => onClick?.()}
-      data-tooltip-id="button-tooltip"
-      data-tooltip-content={tooltip}
-      className={cn(
-        'flex min-h-[36px] items-center justify-center gap-2 rounded-md border border-primary-600 px-5 py-1 font-medium transition-colors hover:bg-primary-200 active:opacity-80 disabled:border-gray-400 disabled:bg-transparent disabled:text-gray-400 disabled:active:opacity-100',
-        ghost && 'border-none bg-transparent hover:bg-primary-100',
-        red &&
-          'border-red-700 text-red-800 hover:bg-red-400/20 hover:text-red-600',
-        filled &&
-          'border-primary-500 bg-primary-500 text-text-50 hover:border-primary-500 hover:bg-primary-700 disabled:border-primary-300 disabled:bg-primary-300 disabled:text-text-700 disabled:hover:border-primary-300 disabled:hover:bg-primary-300',
-        filled &&
-          red &&
-          'border-red-700 bg-red-700 text-red-50 hover:border-red-800 hover:bg-red-600 hover:text-red-50 disabled:border-red-900/80 disabled:bg-red-900/60 disabled:text-red-900 disabled:hover:border-red-900/80 disabled:hover:bg-red-900/60',
-        round && 'rounded-full',
-        className,
-      )}
-      {...props}
-    >
-      {children}
-    </button>
-  );
-};
+const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
+  (
+    {
+      className,
+      variant,
+      size,
+      color,
+      icon,
+      startIcon,
+      endIcon,
+      type = 'button',
+      loading = false,
+      round,
+      asChild = false,
+      ...props
+    },
+    ref,
+  ) => {
+    const Comp = asChild ? Slot : 'button';
+
+    const buttonSize = (size && iconSizes[size]) || iconSizes.md;
+
+    const roundClasses = round ? 'rounded-full' : 'rounded-md';
```

Comment:
I think this should be possible to do as another CVA variant

Reply:
Sizes are now using compoundVariants in CVA.
The two different sizes is if you are using the iconButton component or a start- end icon.